### PR TITLE
New buttons and fetchObject(s) helper

### PR DESCRIPTION
**Pull request by guscost**
Updates Density UI to 17.0.0, and migrates it to use the changed Button component in that release. Also updates dashboard to use a new fetchAllObjects helper with caching.

### CODE REVIEW

File: src/actions/route-transition/admin-user-management-detail.ts [hunk 1]
```diff
@@ -15,9 +16,10 @@ export default function routeTransitionAdminUserManagementDetail(id) {
       setLoading: true,
     });
 
-    let user, spaces, errorThrown = null;
+    let user, hierarchy, spaces, errorThrown = null;
     try {
       user = await accounts().get(`/users/${id}`);
+      hierarchy = await fetchAllObjects<DensitySpaceHierarchyItem>('/spaces/hierarchy');
```
Comment (1egoman) [inline src/actions/route-transition/admin-user-management-detail.ts]: Not related to this PR in particular, but these requests probably should be made in parallel with `Promise.all`.

File: src/actions/route-transition/admin-user-management.ts [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 import fetchAllObjects from '../../helpers/fetch-all-objects';
-import { DensitySpace } from '../../types';
+import { DensitySpace, DensitySpaceHierarchyItem } from '../../types';
 
 export default function routeTransitionAdminUserManagement() {
   return async dispatch => {
     dispatch({ type: ROUTE_TRANSITION_ADMIN_USER_MANAGEMENT });
     const usersResponse = await accounts().get('/users');
-    const hierarchy = (await core().get('/spaces/hierarchy/')).data.map(objectSnakeToCamel);
+    const hierarchy = await fetchAllObjects<DensitySpaceHierarchyItem>('/spaces/hierarchy');
```
Comment (1egoman) [inline src/actions/route-transition/admin-user-management.ts]: Not related to this PR in particular, but these requests probably should be made in parallel with `Promise.all`.

File: src/actions/route-transition/explore-space-data-export.ts [hunk 3]
```diff
@@ -13,7 +12,7 @@ export default function routeTransitionExploreSpaceDataExport(id) {
     dispatch({ type: ROUTE_TRANSITION_EXPLORE_SPACE_DATA_EXPORT, id });
 
     try {
-      const spaceHierarchy = (await core().get('/spaces/hierarchy')).data;
+      const spaceHierarchy = await fetchAllObjects<DensitySpaceHierarchyItem>('/spaces/hierarchy');
       const spaces = await fetchAllObjects<DensitySpace>('/spaces');
       const selectedSpace = spaces.find(s => s.id === id);
```
Comment (1egoman) [inline src/actions/route-transition/explore-space-data-export.ts]: Note related to this PR in particular, but these requests probably should be made in parallel with `Promise.all`.

File: src/actions/route-transition/explore-space-meetings.ts [hunk 4]
```diff
@@ -42,7 +43,7 @@ export default function routeTransitionExploreSpaceMeeting(id, serviceName) {
     // this view unrfortunately.
     let spaces, spaceHierarchy, selectedSpace;
     try {
-      spaceHierarchy = (await core().get('/spaces/hierarchy')).data;
+      spaceHierarchy = await fetchAllObjects<DensitySpaceHierarchyItem>('/spaces/hierarchy');
       spaces = await fetchAllObjects<DensitySpace>('/spaces');
       selectedSpace = spaces.find(s => s.id === id);
```
Comment (1egoman) [inline src/actions/route-transition/explore-space-meetings.ts]: Note related to this PR in particular, but these requests probably should be made in parallel with `Promise.all`.

**Comment by guscost** [comment]
Rob said he approves! QA Passed! Shipping!

**Pull request merged**